hpy/debug/src/debug_ctx_cpython.c

```diff
@@ -35,6 +35,36 @@ static inline PyObject *_dh2py(DHPy dh)
     return _h2py(DHPy_unwrap(dh));
 }
 
+static void _buffer_h2py(HPyContext ctx, const HPy_buffer *src, Py_buffer *dest)
```

Comment:
It would be good to avoid duplicating these helper functions, but I'm not quite sure where we would put them.

Reply:
yes and no. I agree that code duplication is bad, but OTOH this file and `ctx_meth.c` (where the "other" functions live) are supposed to be independent. The mess comes from the fact that the universal mode should be interpreter-agnostic, the debug mode too and both should be independent from each other.
But in practice, we need to put CPython-specific code in both (which is `ctx_meth.c` for the universal mode and `debug_ctx_cpython.c` for the debug mode). Maybe we *could* try to share code between the CPython-specific parts of the two modes, but as long as the code duplication is short&straightforward, I think it's simpler to just live with it.